src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/RequestPayloadValueResolver.php

```diff
@@ -78,6 +78,16 @@ public function resolve(Request $request, ArgumentMetadata $argument): iterable
             throw new \LogicException(sprintf('Mapping variadic argument "$%s" is not supported.', $argument->getName()));
         }
 
+        if ($attribute instanceof MapRequestPayload) {
+            if ('array' === $argument->getType()) {
+                if (!$attribute->type) {
+                    throw new \LogicException(sprintf('For proper deserialization, the %s::$type value is required when the argument type is "array".', MapRequestPayload::class));
```

Comment:
the %s::$type value => `#[MapRequestPayload]'s $type argument`?

Reply:
ah, can we also add the name and the class::method that is missing this?

Reply:
👍